python/pipeline/pupil.py

```diff
@@ -915,42 +879,185 @@ def make(self, key):
+                radius = fit_dict['circle_fit']['radius']
+                visible_portion = fit_dict['circle_visible']['visible_portion']
 
-                data_circle.append([*common_entry, frame_num, center, radius, visible_portion])
+                data_circle.append(
+                    [center, radius, visible_portion])
 
                 # ellipse info
-                center=fit_dict['ellipse_fit']['center']
-                major_radius=fit_dict['ellipse_fit']['major_radius']
-                minor_radius=fit_dict['ellipse_fit']['minor_radius']
-                rotation_angle=fit_dict['ellipse_fit']['rotation_angle']
-                visible_portion=fit_dict['ellipse_visible']['visible_portion']
-
-                data_ellipse.append([*common_entry, frame_num, center, 
-                                            major_radius, minor_radius, 
-                                            rotation_angle, visible_portion])
-        
+                center = fit_dict['ellipse_fit']['center']
+                major_radius = fit_dict['ellipse_fit']['major_radius']
+                minor_radius = fit_dict['ellipse_fit']['minor_radius']
+                rotation_angle = fit_dict['ellipse_fit']['rotation_angle']
+                visible_portion = fit_dict['ellipse_visible']['visible_portion']
+
+                data_ellipse.append([center,
+                                     major_radius, minor_radius,
+                                     rotation_angle, visible_portion])
+
+        data_circle = np.array(data_circle)
+        data_ellipse = np.array(data_ellipse)
+
         # now filter out the outliers by 5.5 std away from mean
-        data_circle = self.filter_by_std(data_circle,fitting_method='circle', std_magnitude=5.5)
-        data_ellipse = self.filter_by_std(data_ellipse,fitting_method='ellipse', std_magnitude=5.5)
+        rejected_ind = DLC_tools.filter_by_std(
+            data=data_circle, fitting_method='circle', std_magnitude=5.5)
+
+        data_circle[rejected_ind] = None, None, -3.0
+
+        common_entry = np.array(list(key.values()))
+        common_matrix = np.tile(common_entry, (nframes, 1))
+
+        data_circle = np.hstack(
+            (common_matrix, np.arange(nframes).reshape(-1, 1), data_circle))
 
         # insert data
         self.Circle.insert(data_circle)
+
+        # now repeat the process for ellipse
+        rejected_ind = DLC_tools.filter_by_std(
+            data=data_ellipse, fitting_method='ellipse', std_magnitude=5.5)
+
+        data_ellipse[rejected_ind, :] = None, None, None, None, -3.0
+
+        data_ellipse = np.hstack(
+            (common_matrix, np.arange(nframes).reshape(-1, 1), data_ellipse))
+
         self.Ellipse.insert(data_ellipse)
 
+
+@schema
+class OnlineMedianFilteredFittedPupil(dj.Computed):
+    definition = """
+    # Fit a circle and an ellipse after filtering.
+    -> Tracking
+    -> filter_config.OnlineMedianFilter
```

Comment:
if you move `OnlineMedianFilter` from pipeline.utils.filter_config, you may need to change the pointer here (no need to change anything in the database, because the tables will still be in the pipeline_eye schema, just the code definition that will change positions)

Reply:
I just commented out this table...  I know now `pupil.py` is dirty, but 1. i want minion jobs to run without any errors 2. I am still unsure if I need this table or not in the future...I will decide within 2-3 weeks